File: Problems/CS50X/Week 2 Arrays/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);
    float L = letters / words * 100;
    float S = sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

    // Compute the Coleman-Liau index

    // Print the grade level
    // printf("%f\n",sentences);
}
int count_letters(string text)

{
    int score = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {

        if (isalpha(text[i]))
        {
            score++;
        }
    }
    return score;
}

int count_words(string text)
{
    // Return the number of words in text
    int score = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isblank(text[i]))
        {
            score++;
        }
    }
    return score + 1;
}

int count_sentences(string text)
{
    int score = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '?' || text[i] == '.' || text[i] == '!')
        {
            score++;
        }
    }
    return score;
}

File: Problems/CS50X/Week 6 Python/sentimental-readability/readability.py
def main():
    main_text = str(input("Text: "))
    letters = float(count_letters(main_text))
    words = float(count_words(main_text))
    sentences = float(count_sentences(main_text))
    L = float(letters / words * 100)
    S = float(sentences / words * 100)
    index = float(0.0588 * L - 0.296 * S - 15.8)
    grade = round(index)
    print(letters,words,sentences)
    if grade<1:
        print("Before Grade 1")
    elif grade>16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")






def count_letters(text):
    text = list(text)
    score = 0
    for i in range(len(text)):
        if text[i].isalpha():
            score+=1
    return score



def count_words(text):
    text = text.split()
    score =0
    for i in range(len(text)):
        if text[i] in text:
            score+=1
    return score


def count_sentences(text):
    text = list(text)
    score = 0
    for i in range(len(text)):
        if str(text[i])=='?'or str(text[i])=='.'or str(text[i])=='!':
            score += 1
    return score

main()
